pkg/jobstore/inmemory/inmemory.go

```diff
@@ -198,19 +214,33 @@ func (d *JobStore) CreateJob(_ context.Context, job model.Job) error {
 		JobID:      job.Metadata.ID,
 		State:      model.JobStateNew,
 		Version:    1,
-		CreateTime: time.Now(),
-		UpdateTime: time.Now(),
+		CreateTime: time.Now().UTC(),
+		UpdateTime: time.Now().UTC(),
 	}
 	d.states[job.Metadata.ID] = jobState
 	d.inprogress[job.Metadata.ID] = struct{}{}
 	d.appendJobHistory(jobState, model.JobStateNew, newJobComment)
+
+	d.triggerEvent(jobstore.JobWatcher, jobstore.CreateEvent, job)
+
+	return nil
+}
+
+// DeleteJob removes a job from storage
+func (d *InMemoryJobStore) DeleteJob(ctx context.Context, jobID string) error {
+	d.triggerEvent(jobstore.JobWatcher, jobstore.DeleteEvent, d.jobs[jobID])
+
```

Comment:
we should trigger the even the deletion

Reply:
I've noticed there's actually an OnCommit() we can use for the transaction to do the notify once we know it'll be successful.  Which feels cleaner than the current way where there is implied knowledge that returning an err means a fail means no notification.